### PR TITLE
Parser fix, only allow ORDER BY ASC and ORDER BY time ASC

### PR DESCRIPTION
**Pull request by gunnaraasen**
Fixes https://github.com/influxdb/influxdb/issues/3380, which was introduced in https://github.com/influxdb/influxdb/pull/2912, which itself was a fix for https://github.com/influxdb/influxdb/issues/2731.

This PR makes the parser accept `ORDER BY ASC` and `ORDER BY time ASC` as the only valid `ORDER BY` clauses. All other `ORDER BY` clauses will return an error.

This also adds some parsing logic for `ORDER BY` clauses and fixes `SortField.String()` to output correctly so the query can be rewritten correctly in the `rewriteSelectStatement` step of the query executor.

### CODE REVIEW

**Comment by otoolep** [comment]
+1

File: influxql/parser.go [hunk 1]
```diff
@@ -1742,10 +1742,22 @@ func (p *Parser) parseOrderBy() (SortFields, error) {
...
 	var fields SortFields
 
+	// If first token is ASC or DESC, all fields are sorted.
+	if tok, pos, lit := p.scanIgnoreWhitespace(); tok == ASC || tok == DESC {
+		if tok == DESC {
+			// Token must be ASC, until other sort orders are supported.
+			return nil, errors.New("only ORDER BY time ASC supported at this time")
+		}
+		return append(fields, &SortField{Ascending: (tok == ASC)}), nil
+	} else if tok != IDENT && tok != STRING {
```
Comment (dgnorton) [inline influxql/parser.go]: Should only accept `IDENT`.  `ORDER BY "foo" ASC` is okay but not `ORDER BY 'foo' ASC`.

File: influxql/parser.go [hunk 2]
```diff
@@ -1777,13 +1794,21 @@ func (p *Parser) parseSortFields() (SortFields, error) {
...
-	// Next token must be ASC, until other sort orders are supported.
-	tok, _, _ := p.scanIgnoreWhitespace()
-	if tok != ASC {
-		return nil, errors.New("only ORDER BY ASC supported at this time")
+	// Scan for IDENT or STRING clause.
+	tok, pos, lit := p.scanIgnoreWhitespace()
+	if tok != IDENT && tok != STRING {
+		return nil, newParseError(tokstr(tok, lit), []string{"identifier"}, pos)
+	}
+	field.Name = lit
```
Comment (dgnorton) [inline influxql/parser.go]: The `// Scan for IDENT...` line through `field.Name = lit` should be changed to...
```
	// Parse field name
	ident, err := p.parseIdent()
	if err != nil {
		return nil, err
	}
	field.Name = ident
```

**Comment by gunnaraasen** [comment]
@dgnorton made the `parseIdent` change.

**Comment by dgnorton** [comment]
+1

**Pull request merged**